src/open_company_web/components/finances/finances_edit.cljs

```diff
@@ -0,0 +1,145 @@
+(ns open-company-web.components.finances.finances-edit
+  (:require [om.core :as om :include-macros true]
+            [om-tools.core :as om-core :refer-macros [defcomponent]]
+            [om-tools.dom :as dom :include-macros true]
+            [open-company-web.router :as router]
+            [open-company-web.lib.utils :as utils]
+            [open-company-web.components.cell :refer [cell]]
+            [open-company-web.api :as api]
+            [open-company-web.dispatcher :refer [app-state]]))
+
+(defcomponent finances-edit-row [data owner]
+  (render [_]
+    (let [prefix (:prefix data)
+          finances-data (:cursor data)
+          is-new (:new finances-data)
+          cell-state (if is-new :new :display)
+          change-cb (:change-cb data)]
+      (dom/tr {}
+        (dom/td {:class "no-cell"} (utils/period-string (:period finances-data) (when is-new :force-year)))
+        (dom/td {}
+          (om/build cell {:value (:cash finances-data)
+                          :placeholder (if is-new "at month end" "")
+                          :prefix prefix
+                          :cell-state cell-state
+                          :draft-cb #(change-cb :cash %)}))
+        (dom/td {}
+          (om/build cell {:value (:revenue finances-data)
+                          :placeholder (if is-new "entire month" "")
+                          :prefix prefix
+                          :cell-state cell-state
+                          :draft-cb #(change-cb :revenue %)}))
+        (dom/td {}
+          (om/build cell {:value (:costs finances-data)
+                          :placeholder (if is-new "entire month" "")
+                          :prefix prefix
+                          :cell-state cell-state
+                          :draft-cb #(change-cb :costs %)}))
+        (dom/td {:class (utils/class-set {:no-cell true :new-row-placeholder is-new})}
+                (if is-new "calculated" (:burn-rate finances-data)))
+        (dom/td {:class (utils/class-set {:no-cell true :new-row-placeholder is-new})}
+                (if is-new "calculated" (:runway finances-data)))))))
+
+(defn save-new-row? [finances-data]
+  (let [new-period (first (filter #(and (contains? % :new) (true? (:new %))) finances-data))]
+    (if (or (not (nil? (:cash new-period)))
```

Comment:
You can use the falsey-ness of nil and truthiness of #'s here to use the more simple:

```clojure
(if (or (:cash new-period) (:costs new-period) (:revenue new-period))
```

And you never want to use:

```clojure
(if (or (...)
   true
   false)
```

in Clojure since or already returns true/false.

So this can be just:

```clojure
(or (:cash new-period) (:costs new-period) (:revenue new-period))
```